created scripts to help automate adding a new subject

henryw30 opened the pull request:
new file:   experiments/githubcloner.sh
	new file:   experiments/henry-parallelization-runner.sh
	new file:   experiments/henry-prioritization-runner.sh
	new file:   experiments/henry-selection-runner.sh
	new file:   experiments/setup.sh

will plan to rename scripts more appropriately

winglam commented on experiments/githubcloner.sh:
```diff
@@ -0,0 +1,41 @@
+echo 'Enter github repository link'
+read LINK 
```
Instead of asking the user to type this information in line by line, I would recommend you to change the script so that you pass all of this information in as arguments when the script is invoked. E.g., bash githubcloner.sh LINK DIR_NAME SHA...
This would allow your script to be usable by other scripts such as one that can clone multiple repositories at once with a predefined list of LINK, DIR_NAME, SHA, ...

henryw30 replied:
I edited the githubcloner to take in arguments now

winglam replied:
Is this file still needed when you have a copy of it in adding-new-experiments/?

winglam commented on experiments/henry-parallelization-runner.sh:
```diff
@@ -0,0 +1,71 @@
+#source ./config.sh
```
Please change these file names to be something more descriptive. henry-X.sh does not inform the reader much about how the script differs from the existing ones. Perhaps something like parallelization-w-arguments.sh is more appropriate.

winglam replied:
Please remove these files if they are no longer needed.

winglam commented on experiments/henry-parallelization-runner.sh:
```diff
@@ -0,0 +1,71 @@
+    #echo 'Running parallelization with resolveDependences and without dependentTestFile for original order'
+    #java -cp $2 edu.washington.cs.dt.impact.Main.OneConfigurationRunner -technique parallelization -order original -origOrder $1-$3-order -testInputDir $5/sootTestOutput-$3 -filesToDelete $1-env-files -project "$4" -testType $3 -numOfMachines $k -outputDir ${initialDir}/${paraDir} -timesToRun ${medianTimes} -resolveDependences ${initialDir}/parallelization-dt-list/"parallelization-$4-$3-$k-origi ...
+    #echo 'Running parallelization with resolveDependences and without dependentTestFile for time order'
+    #java -cp $2 edu.washington.cs.dt.impact.Main.OneConfigurationRunner -technique parallelization -order time -timeOrder $5/$1-$3-time.txt -origOrder $1-$3-order -testInputDir $5/sootTestOutput-$3 -filesToDelete $1-env-files -numOfMachines $k -project "$4" -testType $3 -timesToRun ${medianTimes} -outputDir ${initialDir}/${paraDir} -resolveDependences ${initialDir}/parallelization-dt-list/"paral ...
```
Lines such as these that you are clearly not using should be removed. In other words, remove the lines that you just simply copied over and/or have no intention of using.

winglam commented on experiments/henry-parallelization-runner.sh:
```diff
@@ -0,0 +1,71 @@
+#count=${#experiments[@]}
+
+#while [ "$index" -lt "$count" ]; do
+  echo -e "Starting experiment: $1"
```
You have some unnecessary indentation here.

henryw30 commented:
I renamed the scripts to be a little more descriptive as well as removed some lines that were unused in my scripts. All of them were also moved to their own folder called "adding-new-experiments".

winglam commented on experiments/adding-new-experiments/prioritization-runner-w-arguments.sh:
```diff
@@ -0,0 +1,31 @@
+      java -cp ${3}/*:${4}/*:${5}:${6}:${8} edu.washington.cs.dt.impact.Main.OneConfigurationRunner -technique prioritization -coverage $i -order $j -origOrder $1-$k-order -testInputDir sootTestOutput-$k -filesToDelete $1-env-files -getCoverage -project "${1}" -testType $k -outputDir ${7} -timesToRun 1
+    done
+  done
+  clearTemp $1 $k
```
clearTemp doesn't seem to use arguments.

winglam commented on experiments/adding-new-experiments/selection-runner-w-arguments.sh:
```diff
@@ -0,0 +1,27 @@
+  clearSelectionTemp $8 $2
+done
+
+cd /home/user/dependent-tests-impact/experiments
```
This script doesn't seem to cd anywhere yet you cd to this location at the end. Is this cd necessary?

winglam commented on experiments/adding-new-experiments/selection-runner-w-arguments.sh:
```diff
@@ -0,0 +1,27 @@
+testTypes=(orig auto)
+coverages=(statement function)
+machines=(2 4 8 16)
+priorOrders=(absolute relative)
```
This argument and machines are not used.

winglam replied:
Other scripts seem to have this issue with unused declarations as well.

winglam commented on experiments/adding-new-experiments/prioritization-runner-w-arguments.sh:
```diff
@@ -0,0 +1,31 @@
+  clearTemp $1 $k
+done
+
+clearInstrumentation
```
variableToType.dat sounds like a file that is being created each time OneConfigurationRunner is executed. The only files that should go into clearInstrumentation function should be ones that are generated by soot instrumentation which this script does not seem to do.

winglam commented on experiments/adding-new-experiments/add-new-subject.sh:
```diff
@@ -0,0 +1,170 @@
+# The root directory for the old subject
```
add-new-subject.sh does not seem to describe this file well. The purpose of this file seems to be setting up variables for adding new subjects. Perhaps naming it setup-variables.sh would be more appropriate.

winglam commented on experiments/adding-new-experiments/add-new-subject.sh:
```diff
@@ -0,0 +1,170 @@
+
+
+
+#PRIORITIZATION
```
Is there a reason why the comments below are kept? If not, please delete these extra lines.
